NameLayerMain/src/vg/civcraft/mc/namelayer/group/Group.java

```diff
@@ -395,23 +396,38 @@ public static boolean areLinked(Group supergroup, Group subgroup) {
 		if (supergroup == null || subgroup == null) {
 			return false;
 		}
-		
-		return supergroup.hasSubGroup(subgroup) 
-				&& (subgroup.hasSuperGroup() 
-				&& subgroup.supergroup.equals(supergroup));
+		List <String> names = new LinkedList<String>();
+		names.add(supergroup.getName());
+		//check upwards to prevent cycles
+		if (supergroup.hasSuperGroup()) {
+			names.addAll(getRecursiveNamesUpwards(supergroup));
+		}
+		if (names.contains(subgroup.getName())) {
+			return true;
+		}
+		return false;
+	}
+	
+	private static List <String> getRecursiveNamesUpwards(Group g) {
```

Comment:
Wouldn't this still cycle forever if the supergroup already was part of a cycle?